crates/stellar-scaffold-cli/src/commands/init.rs

```diff
@@ -175,3 +142,34 @@ fn rename_cargo_toml_remove(project: &Path, name: &str) -> Result<(), Error> {
     fs::rename(from, to)?;
     Ok(())
 }
+
+fn clone_repo(repo_url: &str, dest: &Path) -> Result<(), Error> {
+    let status = Command::new("git")
+        .args(["clone", repo_url, dest.to_str().unwrap()])
+        .status()
+        .map_err(|e| Error::GitCloneError(format!("Failed to execute git clone: {e}")))?;
+
+    if !status.success() {
+        return Err(Error::GitCloneError("Git clone command failed".to_string()));
+    }
+    Ok(())
+}
```

Comment:
Is is fine fro now, but I think it's better to use a dep like gitoxide directly so that we don't require an external dep.